playshogi-library-shogi/src/main/java/com/playshogi/library/shogi/models/position/ShogiPosition.java

```diff
@@ -5,13 +5,15 @@
 import com.playshogi.library.shogi.models.Piece;
 import com.playshogi.library.shogi.models.shogivariant.ShogiVariant;
 
-import java.lang.reflect.Array;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Optional;
 
 public class ShogiPosition implements Position<ShogiPosition> {
 
+    // Future work: if moveCount % 2 == 1, it is sente's turn
+    private int moveCount;
+    @Deprecated
     private boolean senteToPlay;
```

Comment:
I'm not totally sure about deprecating the "senteToPlay" - it seems simpler to keep this independent of the moveCount, to allow for games where for some reason moveCount % 2 == 1.

I am thinking here of 2 edge cases:
- Handicap game, where the player at the bottom of the board moves second (which I call "sente" but should really be "Black", or "senteOrShitate")
- For lessons etc, there will be special "moves" in the game tree that do not change the turn to play - I'm not totally sure how that would fit there.

I'd suggest to simply keep both variables without deprecation for now.

Reply:
OK, I suppose my concern was that perhaps someday we choose to save move sequences as kifu or PSN, but on second thought that should be fine... I'll remove the deprecation tag.

I would expect that (similar to PGN starting with "1..." instead of "1.") if White moves first, the move counter would be even ("2." instead of "1"), unless existing kifu or PSN already support a contrary standard.